pandas/plotting/_core.py

```diff
@@ -1714,13 +1714,28 @@ def _plot(data, x=None, y=None, subplots=False,
                 data = data.set_index(x)
 
             if y is not None:
-                if is_integer(y) and not data.columns.holds_integer():
+                int_cols = is_integer(y) or any(is_integer(col) for col in y)
+                if int_cols and not data.columns.holds_integer():
                     y = data.columns[y]
-                elif not isinstance(data[y], ABCSeries):
-                    raise ValueError("y must be a label or position")
-                label = kwds['label'] if 'label' in kwds else y
-                series = data[y].copy()  # Don't modify
-                series.name = label
+                elif not isinstance(data[y], (ABCSeries, ABCDataFrame)):
+                    raise ValueError(
+                        "y must be a label or position or list of them"
+                    )
+
+                label_kw = kwds['label'] if 'label' in kwds else False
+                new_data = data[y].copy()  # Don't modify
+
+                if isinstance(data[y], ABCSeries):
+                    label_name = label_kw or y
+                    new_data.name = label_name
+                else:
+                    match = is_list_like(label_kw) and len(label_kw) == len(y)
+                    if label_kw and not match:
+                        raise ValueError(
```

Comment:
Can you add a test that raises this assertion?

Reply:
done

Reply:
where is this test?

Reply:
https://github.com/MasonGallo/pandas/blob/d6d824f3097f2ba48f8778d65422d8471a9f746f/pandas/tests/plotting/test_frame.py#L2176